### PR TITLE
Add missing sound features from Scratch

### PR DESCRIPTION
**Pull request by djdolphin**
This adds instruments, drums, volume blocks, and the loudness block. It seems to work in every major browser except for mobile Safari. I still have no idea why. :(

I've moved sound-related code to a separate file since objects.js is already *really* long. I haven't done the header because there seems to be a pretty specific format, and I don't want to mess it up. :)

### CODE REVIEW

File: gui.js [hunk 1]
```diff
@@ -5009,22 +5023,28 @@ IDE_Morph.prototype.setCloudURL = function () {
...
-        request.open('GET', url, false);
-        request.send();
+
+    if (!errorCall) {
+        errorCall = function () {
+            myself.showMessage('Error: unable to retrieve ' + url);
+        };
+    }
+
+    request.open('GET', url, true);
```
Comment (cycomachead) [inline gui.js]: Shouldn't we just leave of `true` since it's the default mode?

**Comment by cycomachead** [comment]
WooooooHooo!!

Async loading! Yes!

Minor comments from skimming:
- [ ] The default block comment should be at the top of the sounds.js
- [ ] `soundbank/` → `Sounds/` ? (keeps with the rest of our folders)



> except for mobile Safari. I still have no idea why. :(

Unfortunately, we have significant limitations in Safari to deal with. AFAIK, Mobile Safari will only play audio as a direct result of a user action.

**Comment by jmoenig** [comment]
Aw, man. A giant PR that bundles lot of individual changes into one big package that I have to either accept or reject in total. Glancing cursorily over the code it looks nice, but this is going to keep me tied up forever, once I've managed to catch up with all the retina issues that still keep surfacing and chasing other issues that also keep coming up. Couldn't this have been broken up into smaller pieces? Couldn't asynchronous resource loading have been parcelled out, same as microphone support and volume? The way it is I won't be able to pull this for a long while, because it's going to draw all my attention to follow-up issues :-(

**Comment by cycomachead** [comment]
> Couldn't this have been broken up into smaller pieces?

Yeah, this would actually be really easy to tease apart. At least, the async stuff.

File: objects.js [hunk 2]
```diff
@@ -2780,6 +2826,23 @@ SpriteMorph.prototype.reportSounds = function () {
     return this.sounds;
 };
 
+// SpriteMorph volume
+
+SpriteMorph.prototype.setVolume = function (volume) {
+    if (!isNaN(volume)) {
```
Comment (dekrain) [inline objects.js]: Shouldn't be `if (!isNaN(+volume)) {` ?

**Comment by djdolphin** [comment]
Sure thing, I'll separate these and address those other things in the new pull requests. They should probably be added in this order, since they sort of depend on each other:
- the async stuff
- instruments and drums
- volume and/or loudness.

**Comment by cycomachead** [comment]
Tagging #235 

https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/issues/235#issuecomment-235678875

**Pull request closed without merging**